api/libproject/devstart.rb

```diff
@@ -276,15 +277,24 @@ def drop_cloud_cdr(*args)
   end
 end
 
+def start_cloud_sql_proxy()
+  common = Common.new
+  Common.new.run_inline %W{docker-compose up -d cloud-sql-proxy}
+  at_exit { common.run_inline %W{docker-compose down} }
+end
+
 def connect_to_cloud_db(*args)
-  GcloudContext.new("connect-to-cloud-db", args, true).run do |ctx|
-    pw = ENV["WORKBENCH_DB_PASSWORD"]
-    # TODO Switch this to run_inline once Common supports redaction.
-    run_with_redirects(
-        "mysql -u \"workbench\" -p\"#{pw}\" --host 127.0.0.1 "\
-        "--port 3307 --database #{ENV["DB_NAME"]}",
-        to_redact=pw)
-  end
+  Credentials.new.maybe_download_db_vars :dev
```

Comment:
Are you actually copying the database credentials to db/vars.dev.env here? I don't think we want to do that. 

We use GcloudContext with use_cloudsql_proxy=True to take care of copying DB creds into a temp file, running Cloud SQL proxy, reading the DB vars into ENV, and cleaning up everything when done...

We should keep using GcloudContext here. If you want to avoid having MySQL installed locally, just change the run_with_redirects command to use docker-compose run, and leave everything else the way it was. Doesn't that work?

Reply:
(Or if the issue is you need to run cloud-sql-proxy in docker, too, then change that as well in GcloudContext... make everything dockerized.)

Reply:
(And if this is too much of a pain... then just install MySQL client. It's very easy. :)

Reply:
I agree it seems like there's some duplication of effort here.

I do like avoiding the developer need for MySQL client. (it's easy to install *some* version, but it turned out to be quite a pain to install the *right* version on Google workstations. And anyway encapsulating/minimizing deps is nice.)

Reply:
Maybe an implicit followup here is to either improve docs around GcloudContext, or do a quick once-over in our tech talk slot.

Reply:
See how this new approach aligns with your thoughts.